UnityGLTF/Assets/UnityGLTF/Scripts/Extensions/GLTFSceneExporter.Animation.cs

```diff
@@ -0,0 +1,335 @@
+﻿using System;
+using System.Collections.Generic;
+using GLTF.Schema;
+using UnityEngine;
+using UnityEditor.Animations;
+using UnityEditor;
+using System.Linq;
+
+namespace UnityGLTF
+{
+	public partial class GLTFSceneExporter
+	{
+		private enum Property
+		{
+			m_LocalPosition = GLTFAnimationChannelPath.translation,
+			m_LocalRotation = GLTFAnimationChannelPath.rotation,
+			m_LocalScale = GLTFAnimationChannelPath.scale,
+			blendShape = GLTFAnimationChannelPath.weights,
+		}
+
+		//holds already exported animation clips
+		private readonly Dictionary<object, int> _animationCache = new Dictionary<object, int>();
+
+		private void ExportAnimations(IEnumerable<Transform> transforms)
+		{
+			foreach (var transform in transforms)
+			{
+				//use Animator instead of Animation for support of newest features
+				var gameObject = transform.gameObject;
+				foreach (var animator in gameObject.GetComponentsInChildren<Animator>())
+				{
+					var animatorController = (AnimatorController)animator.runtimeAnimatorController;
+					if (animatorController == null)
+					{
+						//no animations attached
+						continue;
+					}
+
+					foreach (var layer in animatorController.layers)
+					{
+						var animationStates = layer.stateMachine.states;
+						foreach (var childState in animationStates)
+						{
+							var state = childState.state;
+							if (state.motion is BlendTree)
+							{
+								throw new NotSupportedException();
+							}
+
+							var animationClip = (AnimationClip)state.motion;
+							if (animationClip != null)
+							{
+								//export animation clip from each state of the state machine
+								this.ExportAnimation(gameObject, animationClip);
```

Comment:
this. isn't needed

Reply:
We should put up a coding convention document so contributors know what to use.